zetaclient/chains/bitcoin/tokenizer.go

```diff
@@ -0,0 +1,156 @@
+package bitcoin
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"github.com/btcsuite/btcd/txscript"
+)
+
+func newScriptTokenizer(script []byte) scriptTokenizer {
+	return scriptTokenizer{
+		script: script,
+		offset: 0,
+	}
+}
+
+// scriptTokenizer is supposed to be replaced by txscript.ScriptTokenizer. However,
+// it seems currently the btcsuite version does not have ScriptTokenizer. A simplified
+// version of that is implemented here. This is fully compatible with txscript.ScriptTokenizer
+// one should consider upgrading txscript and remove this implementation
+type scriptTokenizer struct {
+	script []byte
+	offset int
+	op     byte
+	data   []byte
+	err    error
+}
+
+// Done returns true when either all opcodes have been exhausted or a parse
+// failure was encountered and therefore the state has an associated error.
+func (t *scriptTokenizer) Done() bool {
```

Comment:
Consideration for all the scriptTokenizer methods: are we going to use them outside the `bitcoin` package? If that's not the case we can simply make them private.

Reply:
I would say keep it as it is for now. Ideally we should be using the bitcoin script functionalities from btcsuite. The current version in the code base does not have these functionalities. But higher versions do. In this case, we should keep the method name as of now and once versions are upgraded, just swap to the library implementation.

Reply:
Makes sense, thanks!